packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -198,10 +211,14 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
 
   React.useEffect(() => {
     setGridState((state) => {
+      if (selectionModel) {
+        return state;
+      }
```

Comment:
The two changes in this useEffect looks KO. The behavior should be controlled **invariant**. Meaning, if the uncontrolled selection unselects when the `rows` changes, then the controlled version of the selection should behave identically (unselect).

Reply:
Here, we stop unselecting if controlled 💥